src/wp-includes/formatting.php

```diff
@@ -2035,7 +2035,15 @@ function sanitize_file_name( $filename ) {
 	}
 
 	if ( $utf8_pcre ) {
-		$filename = preg_replace( "#\x{00a0}#siu", ' ', $filename );
+		/**
+		 * Replace all whitespace characters with a basic space (U+0020).
+		 *
+		 * Characters in the White_Space category are listed with “Zs” in
+		 * their entry in the UnicodeData file maintained at the linked URL.
+		 *
+		 * @see https://www.unicode.org/Public/UCD/latest/ucd/UnicodeData.txt
+		 */
+		$filename = preg_replace( '#\p{Zs}#siu', ' ', $filename );
```

Comment:
The `u` (unicode) flag is clearly necessary here. I'm curious about `s` (single line) and `i` (case insensitive) which don't seem to serve any purpose. Do you have intuition about why they'd be here?

Reply:
Not at all. I noted some quirks as well but this patch proposal is about a minimal change to resolve the known common issue so I left them unchanged, for later work to clean up or propose more robust improvements.